dev-libs/libprelude/libprelude-4.0.0.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6}} )
+
+USE_RUBY="ruby21 ruby22 ruby23 ruby24"
+
+inherit autotools python-r1 ruby-single
+
+DESCRIPTION="Prelude-SIEM Framework Library"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc lua python ruby perl"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="net-libs/gnutls:=
+	perl? ( dev-lang/perl:= virtual/perl-ExtUtils-MakeMaker )
+	lua? ( dev-lang/lua:* )
+	ruby? ( ${RUBY_DEPS} )
+	python? ( ${PYTHON_DEPS} )"
+
+DEPEND="${RDEPEND}
+	doc? ( dev-util/gtk-doc )
```

Comment:
with USE=-doc, gtk docs are not installerd because in ./configure we have --disable-gtk-doc

Did I miss something ?

Reply:
You've missed the fact that they actually are installed.

```
$ tar -tf /srv/nfs/common/packages/athlon64/dev-libs/libprelude/libprelude-4.0.0-1.xpak  | grep gtk-doc
./usr/share/gtk-doc/
./usr/share/gtk-doc/html/
./usr/share/gtk-doc/html/libprelude/
./usr/share/gtk-doc/html/libprelude/api-index-full.html
./usr/share/gtk-doc/html/libprelude/ch01.html
./usr/share/gtk-doc/html/libprelude/deprecated-api-index.html
./usr/share/gtk-doc/html/libprelude/index.html

Reply:
indeed but it shouldn't, I fixed it in next push

Reply:
It is common for gtk-doc packages to install gtk-doc docs without any flags. Don't work around that without a very good reason to.

Reply:
there is a USE flag and the configure have the option so I think it has to be fixed.

I will push the patch upstream. If in configure we provide --disable-gtk-doc, gtk-doc do not have to be installed, with or without a useflag

Reply:
Do not conflate USE flags and configure options unnecessary. Just because upstream provides a configure option doesn't mean we have to provide a USE flag.

It is normal for gtk-doc to provide such a flag to control using gtk-doc builder. It is useful when building out of git (or snapshot) where normally gtk-doc would be required to build documentation.

However, for release tarballs (made using `make dist`) the prebuilt documentation is already included and so gtk-doc is unnecessary. Therefore, the configure option does nothing.

Reply:
So what did you expect ? Do you prefere that I remove my patch and the USE doc ?
On other linux distribution, docs (included gtk-doc) are in another package and not required. My patch is in this way.

Reply:
Yes, that would be preferred. There is a major difference between having a separate package, and having USE flags to unnecessarily rebuild this package for something that normally works out of the box.

Reply:
too bad that people are obliged to have gtk-doc installed when they just need the library but ok, you are the reviewer.

Note : I do not talk about rebuild the gtk-doc, I talk about installed it

Reply:
Didn't I tell you at least twice that you *do not need gtk-doc installed* to install doc files from release tarball?

Reply:
Yes I write it too quickly sorry (I though about it in my car) ... I didn't want to say "have gtk-doc installed" but "have the prelude gtk documentation installed" (so not the dependence)

The initial discussion was about having or not the USE flag "doc". For me, it was for installing or not the gtk documentation. For you, if I well understand, it was rebuilding the gtk documentation. You are right about the dependence that is not needed because the documentation is already in the release tarball. But I was not talking about this. I talk about install it.

The situation is that if we want to not automatically install the gtk documentation, we need variables from the configure and these variables are only set if the gtk-doc dependence is ok (even if it is not necessary)

So, for the end user, if he want to control if the gtk-documentation is installed or not, he need to control the configure flag, so he need the gtk-doc dependence and the USE "doc". This was because I did it.

As I said, I will remove my proposition about this to do it in your way.

Reply:
done